content/algorithms/planarity/planarity.md

```diff
@@ -0,0 +1,939 @@
+        else:
+            return False, None
+    else:
+        # graph is planar
+        return True, embedding
+```
+
+
+```python
+#Example 1
+# Load GraphML file
+G = nx.read_graphml("./data/planar_graph.graphml")
+
+# Draw the graph
+pos = nx.planar_layout(G)  # Planar layout for better visualization
+plt.figure(figsize=(6, 6))
+nx.draw(G, pos, with_labels=True, node_color='lightcoral', edge_color='black', node_size=500, font_size=12)
+plt.title("Planar Graph from GraphML")
+plt.show()
+```
+
+```python
+#Example 2
+# Load GraphML file
+G = nx.read_graphml("./data/non_planar_k3_3.graphml")
+
+# Draw the graph
+pos = nx.spring_layout(G)  # Spring layout for visualization
+plt.figure(figsize=(6, 6))
+nx.draw(G, pos, with_labels=True, node_color='lightcoral', edge_color='black', node_size=500, font_size=12)
+plt.title("Non-Planar Graph (K3,3) from GraphML")
+plt.show()
+```
+
+## Check Planarity Recusrivly
+The function check_planarity_recursive recursively checks whether a graph G is planar. It uses a method lr_planarity_recursive from the LRPlanarity class to determine planarity. If the graph is planar, it returns True along with a PlanarEmbedding. If the graph is non-planar, it either returns False and a counterexample (if the counterexample=True flag is set) or just False with no counterexample. The recursive nature of the function allows it to break down the problem in smaller steps, examining the graph’s structure in a depth-first manner to assess its planarity.
+
+---
+
+
+
+
+```python
+def check_planarity_recursive(G, counterexample=False):
+    """Recursive version of :meth:`check_planarity`."""
+    planarity_state = LRPlanarity(G)
+    embedding = planarity_state.lr_planarity_recursive()
+    if embedding is None:
+        # graph is not planar
+        if counterexample:
+            return False, get_counterexample_recursive(G)
+        else:
+            return False, None
+    else:
+        # graph is planar
+        return True, embedding
+```
+
+
+```python
+# Example 1: Planar Graph
+# Load GraphML file
+G = nx.read_graphml("./data/planar_graph.graphml")
+
+# Draw the graph
+pos = nx.planar_layout(G)  # Planar layout for better visualization
+plt.figure(figsize=(6, 6))
+nx.draw(G, pos, with_labels=True, node_color='lightcoral', edge_color='black', node_size=500, font_size=12)
+plt.title("Planar Graph from GraphML")
+plt.show()
+```
+
+## Notes
+
+**Embedding** - A (combinatorial) embedding consists of cyclic orderings of the incident edges at each vertex. Given such an embedding there are multiple approaches discussed in literature to drawing the graph (subject to various constraints, e.g. integer coordinates), see e.g. [2].
+
+
+
+The planarity check algorithm and extraction of the combinatorial embedding is based on the Left-Right Planarity Test [1].
+
+
+
+A counterexample is only generated if the corresponding parameter is set, because the complexity of the counterexample generation is higher.
```

Comment:
Even if the complexity was the same, it would take significant time so the reason for making it optional is really about it taking time to do, the complexity isn’t as important as the extra time it takes. (which is related to the complexity, but those “constants” in the complexity description mean we can’t base time on complexity directly.